files/rsyslog/rsyslog.sh

```diff
@@ -6,26 +6,115 @@ export LOGGING_FILE_PATH=${LOGGING_FILE_PATH:-"/var/log/rsyslog/rsyslog.log"}
+        elif [ "${USE_MMEXTERNAL,,}" = "true" ]; then
+            export SKIP_EMPTY=false
+            export USE_MMEXTERNAL=true
+            echo "INFO: USE_MMEXTERNAL was set but none of the UNDEFINED fields were set. Consider using SKIP_EMPTY=true instead."
+        else
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+            echo "INFO: SKIP_EMPTY was set but USE_MMEXTERNAL was set to false. Enabling SKIP_EMPTY."
+        fi
+    elif [ "${SKIP_EMPTY,,}" = "true" ]; then
+        if [ "${USE_MMEXTERNAL:-}" = "" ]; then
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+        elif [ "${USE_MMEXTERNAL,,}" = "true" ]; then
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+            echo "INFO: Both USE_MMEXTERNAL and SKIP_EMPTY were set to true. Respecting SKIP_EMPTY."
+        else
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+        fi
+    else
+        if [ "${USE_MMEXTERNAL:-}" = "" ]; then
+            export SKIP_EMPTY=false
+            export USE_MMEXTERNAL=true
+            echo "INFO: SKIP_EMPTY was set to false but none of the UNDEFINED fields were set. Consider using SKIP_EMPTY=true instead."
+        elif [ "${USE_MMEXTERNAL,,}" = "true" ]; then
+            export SKIP_EMPTY=false
+            export USE_MMEXTERNAL=true
+            echo "INFO: SKIP_EMPTY was set to false but none of the UNDEFINED fields were set. Consider using SKIP_EMPTY=true instead."
+        else
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+            echo "INFO: Both USE_MMEXTERNAL and SKIP_EMPTY were set to false. Enabling SKIP_EMPTY."
+        fi
+    fi
+else
+    if [ "${SKIP_EMPTY:-}" = "" ]; then
+        if [ "${USE_MMEXTERNAL:-}" = "" ]; then
+            export SKIP_EMPTY=false
+            export USE_MMEXTERNAL=true
+        elif [ "${USE_MMEXTERNAL,,}" = "true" ]; then
+            export SKIP_EMPTY=false
+            export USE_MMEXTERNAL=true
+        else
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+            echo "INFO: SKIP_EMPTY was not set and USE_MMEXTERNAL was set to false but the UNDEFINED fields were configured. Enabling SKIP_EMPTY. Consider using USE_MMEXTERNAL=true instead."
+        fi
+    elif [ "${SKIP_EMPTY,,}" = "true" ]; then
+        if [ "${USE_MMEXTERNAL:-}" = "" ]; then
+            # default - mmnormalize skip-empty
+            export SKIP_EMPTY=true
+            export USE_MMEXTERNAL=false
+            echo "INFO: SKIP_EMPTY was set to true and USE_MMEXTERNAL was not set but the UNDEFINED fields were configured. Consider using USE_MMEXTERNAL=true instead."
+        elif [ "${USE_MMEXTERNAL,,}" = "true" ]; then
+            export SKIP_EMPTY=false
+            export USE_MMEXTERNAL=true
+            echo "INFO: Both USE_MMEXTERNAL and SKIP_EMPTY were set to true. Respecting USE_MMEXTERNAL."
```

Comment:
I think if the user sets `SKIP_EMPTY=true` that should take precedence, even if the user has set both `USE_MMEXTERNAL=true` and the UNDEFINED field configuration.

Reply:
Hmmm, but doesn't s/he expect to process logs with UNDEFINED configurations?

Reply:
But `SKIP_EMPTY=true` and the precedent is to honor the `SKIP_EMPTY=true` setting over everything else.